dask/highlevelgraph.py

```diff
@@ -1271,13 +1320,52 @@ def to_graphviz(
         attrs.setdefault("fontsize", str(node_size))
         attrs.setdefault("tooltip", str(node_tooltips))
 
+        if color == "layer_type":
+            layer_colors = {
+                "DataFrameIOLayer": "#CCC7F9",  # purple
+                "ShuffleLayer": "#F9CCC7",  # rose
+                "SimpleShuffleLayer": "#F9CCC7",  # rose
+                "ArrayOverlayLayer": "#FFD9F2",  # pink
+                "BroadcastJoinLayer": "#D9F2FF",  # blue
+                "Blockwise": "#D9FFE6",  # green
+                "BlockwiseLayer": "#D9FFE6",  # green
+                "BlockwiseCreateArray": "#D9FFE6",  # green
+                "MaterializedLayer": "#DBDEE5",  # gray
+            }
+
+            node_color = layer_colors.get(layer_type)
+
+            attrs.setdefault("fillcolor", str(node_color))
+            attrs.setdefault("style", "filled")
+
         g.node(layer_name, **attrs)
 
     for layer, deps in hg.dependencies.items():
         layer_name = name(layer)
         for dep in deps:
             dep_name = name(dep)
             g.edge(dep_name, layer_name)
+
+    if color == "layer_type":
+        legend_title = "Key"
+        legend_label = (
+            '<<TABLE BORDER="0" CELLBORDER="1" CELLSPACING="0" CELLPADDING="5">'
+            "<TR><TD><B>Legend: Layer types</B></TD></TR>"
+            '<TR><TD BGCOLOR="#CCC7F9">DataFrame IO</TD></TR>'
```

Comment:
Generate this list of <TR> from `layer_colours` to eliminate repetition? Should we only show those items that actually feature in the graph?

Reply:
Yes, that seems a like a neat idea :rocket:

Reply:
![image](https://user-images.githubusercontent.com/62539811/129739411-fddc4483-c95e-4c10-a10b-1ccecac6142c.png)
The problem here is that since I have selected the same color for all Blockwise Layers. it would show up twice

Reply:
Is this ok? or do I add extra checks so it only shows the same color Layers once
in this case, it would skip showing `BlockwiseCreateArray`

Reply:
@martindurant

Reply:
Up to you, I don't insist.

Reply:
It works otherwise. 
![image](https://user-images.githubusercontent.com/62539811/129739975-dc2af518-ab30-47ae-9e15-2b320297c1b7.png)

Reply:
I was thinking of making the `BlockwiseLayer` and `BlockwiseCreateArray` a little darker shades of green. How does that sound?
Same goes with Shuffle and SimpleShuffle layers. Same color but a little darker shade to differentiate.

Reply:
I suggest that we don't want too many colours

Reply:
Oh cool. In that case, I will leave it as it is.
:rocket: 

This is actually much better than showing the entire Legend every time. Great suggestion :smile: